### PR TITLE
Provide define for maximum map edge size

### PR DESCRIPTION
**Pull request by janisozaur**

### CODE REVIEW

**Comment by Margen67** [comment]
Can this go beyond 254/256?

**Comment by janisozaur** [comment]
no

**Comment by Broxzier** [comment]
Pinning #5282 as this checks for the hardcoded practical size.

File: src/openrct2/world/map.c [hunk 1]
```diff
@@ -471,7 +471,7 @@ void map_strip_ghost_flag_from_elements()
 	rct_map_element *mapElement = gMapElements;
 	do {
 		mapElement->flags &= ~MAP_ELEMENT_FLAG_GHOST;
-	} while (++mapElement < gMapElements + 0x30000);
+	} while (++mapElement < gMapElements + 3 * (MAX_MAP_EDGE_SIZE * MAX_MAP_EDGE_SIZE));
```
Comment (jensj12) [inline src/openrct2/world/map.c]: Why not use `MAX_TILE_MAP_ELEMENT_POINTERS`?

File: src/openrct2/world/map.c [hunk 2]
```diff
@@ -4573,8 +4573,8 @@ bool map_surface_is_blocked(sint16 x, sint16 y){
 /* Clears all map elements, to be used before generating a new map */
 void map_clear_all_elements()
 {
-	for (sint32 y = 0; y < (256 * 32); y += 32) {
-		for (sint32 x = 0; x < (256 * 32); x += 32) {
+	for (sint32 y = 0; y < (MAX_MAP_EDGE_SIZE * 32); y += 32) {
+		for (sint32 x = 0; x < (MAX_MAP_EDGE_SIZE * 32); x += 32) {
```
Comment (IntelOrca) [inline src/openrct2/world/map.c]: `(MAX_MAP_EDGE_SIZE * 32)` could be its own constant too

**Comment by IntelOrca** [comment]
`0x30000` should be MAX_MAP_ELEMENTS, its not really related to the size of the map.

**Comment by Gymnasiast** [comment]
Why did you define a new constant, considering we already have `MAXIMUM_MAP_SIZE_TECHNICAL`?

**Comment by Gymnasiast** [comment]
Are you planning on finishing this?

**Comment by janisozaur** [comment]
Yes, I'm going to do it after the SDL PR

**Comment by janisozaur** [comment]
@Gymnasiast `MAXIMUM_MAP_SIZE_TECHNICAL` is really just used for `mapgen.c` file and it fits in line with other defines, so I left it be.

This PR is rebased + updated.

File: src/openrct2/windows/map.c [hunk 3]
```diff
@@ -176,7 +176,7 @@ static uint8 _activeTool;
 static uint32 _currentLine;
 
 /** rct2: 0x00F1AD68 */
-static uint8 (*_mapImageData)[512][512];
```
Comment (marijnvdwerf) [inline src/openrct2/windows/map.c]: Not necessarily agreeing with these changes. 512x512 is the canvas size, and not directly related to the map size.

Comment (janisozaur) [inline src/openrct2/windows/map.c]: I remember it crashed when this was not updated (and I changed the defined value)

**Comment by Gymnasiast** [comment]
I find MAXIMUM_MAP_SIZE_TECHNICAL a better description and would prefer it to be used throughout.

File: src/openrct2/windows/map.c (hunk 3 above)
Comment (Gymnasiast) [inline src/openrct2/windows/map.c]: Would it be an idea to put a comment in explaining Marijn's point and leave @janisozaur 's code as it is?

File: src/openrct2/windows/map.c [hunk 4]
```diff
@@ -1711,16 +1706,16 @@ static void map_window_set_pixels(rct_window *w)
...
 }
 
 static void map_window_screen_to_map(sint32 screenX, sint32 screenY, sint32 *mapX, sint32 *mapY)
 {
 	sint32 x, y;
 
-	screenX = ((screenX + 8) - 256) / 2;
-	screenY = ((screenY + 8)      ) / 2;
+	screenX = ((screenX + 8) - MAXIMUM_MAP_SIZE_TECHNICAL) / 2;
+	screenY = ((screenY + 8)                    ) / 2;
```
Comment (Gymnasiast) [inline src/openrct2/windows/map.c]: This whitespace looks weird.

Comment (janisozaur) [inline src/openrct2/windows/map.c]: It's a leftover from previous version. I'll just remove it.

File: src/openrct2/windows/map.c (hunk 3 above)
Comment (IntelOrca) [inline src/openrct2/windows/map.c]: Just make a new constant which equates to `MAXIMUM_MAP_SIZE_TECHNICAL * 2`

**Pull request merged**